myia/anf_ir.py

```diff
@@ -16,11 +16,35 @@
 
 from myia.ir import Node
 from myia.utils import Named
+import json
 
 PARAMETER = Named('PARAMETER')
 APPLY = Named('APPLY')
 
 
+class Debug(types.SimpleNamespace):
+    """Debug information for an object.
+
+    Attributes:
+        name: The name of the object.
+    """
+    __curr_id__ = 0
```

Comment:
PEP8 on leading and trailing underscores: "Never invent such names; only use them as documented."

Reply:
Also, do we really need name mangling here?

Reply:
If we want unique names for nodes that have no name, yes.

Reply:
Actually, `__curr_id__` doesn't get mangled. `__curr_id` would get mangled, but I don't see how that has anything to do with guaranteeing uniqueness. It simply renames the attribute from `__curr_id` to `_Debug__curr_id`.

Reply:
Oh, that's what you meant. Yeah I don't particularly care whether it's mangled or not.